src/triage/risklist/__init__.py

```diff
@@ -0,0 +1,171 @@
+        query=model_info['cohort_config']['query'],
+        entity_date_table_name=cohort_table_name
+    )
+    cohort_table_generator.generate_entity_date_table(as_of_dates=[dt_from_str(as_of_date)])
+
+    # 3. Generate feature aggregations
+    feature_generator = FeatureGenerator(
+        db_engine=db_engine,
+        features_schema_name="production",
+        feature_start_time=model_info['feature_start_time'],
+    )
+    collate_aggregations = feature_generator.aggregations(
+        feature_aggregation_config=model_info['feature_config'],
+        feature_dates=[as_of_date],
+        state_table=cohort_table_name
+    )
+    feature_generator.process_table_tasks(
+        feature_generator.generate_all_table_tasks(
+            collate_aggregations,
+            task_type='aggregation'
+        )
+    )
+
+    # 4. Reconstruct feature disctionary from feature_names and generate imputation
+    reconstructed_feature_dictionary = {}
+    imputation_table_tasks = OrderedDict()
+    with db_engine.begin() as conn:
+        for aggregation in collate_aggregations:
+            feature_prefix = aggregation.prefix
+            feature_group = aggregation.get_table_name(imputed=True).split('.')[1].replace('"', '')
+            feature_names_in_group = [f for f in model_info['feature_names'] if re.match(f'\A{feature_prefix}', f)]
+            reconstructed_feature_dictionary[feature_group] = feature_names_in_group
+
+            # Make sure that the features imputed in training should also be imputed in production
+            features_imputed_in_train = [f for f in set(feature_names_in_group) if f + '_imp' in feature_names_in_group]
+            results = conn.execute(aggregation.find_nulls())
+            null_counts = results.first().items()
+
+            features_imputed_in_production = [col for (col, val) in null_counts if val > 0]
+            total_impute_cols = set(features_imputed_in_production) | set(features_imputed_in_train)
+            total_nonimpute_cols = set(f for f in set(feature_names_in_group) if '_imp' not in f) - total_impute_cols
+            task_generator = feature_generator._generate_imp_table_tasks_for
+            imputation_table_tasks.update(task_generator(
+                aggregation,
+                impute_cols=list(total_impute_cols),
+                nonimpute_cols=list(total_nonimpute_cols)
+                )
+            )
+    feature_generator.process_table_tasks(imputation_table_tasks)
+
+    # 5. Build matrix
+    db_config = {
+        "features_schema_name": "production",
+        "labels_schema_name": "public",
+        "cohort_table_name": cohort_table_name,
+    }
+
+    matrix_builder = MatrixBuilder(
+        db_config=db_config,
+        matrix_storage_engine=matrix_storage_engine,
+        engine=db_engine,
+        experiment_hash=None,
+        replace=True,
+    )
+
+    matrix_metadata = {
+        'as_of_times': [as_of_date],
+        'matrix_id': str(as_of_date) + '_prediction',
```

Comment:
How about we change 'prediction' to 'risklist' and include the model id in this 'matrix_id'?

Reply:
@tweddielin